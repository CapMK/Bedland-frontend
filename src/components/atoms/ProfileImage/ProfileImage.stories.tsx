import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Avatar from 'assets/img/Avatar.svg';
import ProfileImage from './ProfileImage';

const meta = {
  title: 'Atoms/ProfileImage',
  component: ProfileImage,
  tags: ['autodocs'],
  argTypes: {
    src: {
      table: {
        disable: true,
      },
    },
    size: {
      description: 'Size of the image in rem',
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof ProfileImage>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    src: Avatar,
    size: 1.5,
  },
};
