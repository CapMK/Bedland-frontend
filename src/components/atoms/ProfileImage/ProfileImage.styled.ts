import styled from 'styled-components';

export const Profile = styled.img<{ size: number }>`
  border-radius: 50%;
  height: ${({ size }) => `${size}rem`};
  width: ${({ size }) => `${size}rem`};
`;

export default Profile;
