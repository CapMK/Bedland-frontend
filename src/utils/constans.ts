export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
