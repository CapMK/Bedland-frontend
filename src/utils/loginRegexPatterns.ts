import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
