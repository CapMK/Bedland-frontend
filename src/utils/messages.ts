const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
