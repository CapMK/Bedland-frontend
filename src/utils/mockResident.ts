const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
