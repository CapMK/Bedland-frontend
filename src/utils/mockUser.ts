import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
