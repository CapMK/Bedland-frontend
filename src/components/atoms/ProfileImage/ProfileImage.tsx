import * as Styled from './ProfileImage.styled';

type ProfileImageProps = {
  src: string;
  size: number;
};

const ProfileImage = ({ src, size }: ProfileImageProps) => (
  <Styled.Profile src={src} alt='profile' size={size} />
);

export default ProfileImage;
