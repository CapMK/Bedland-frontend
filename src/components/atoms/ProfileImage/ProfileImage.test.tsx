import Avatar from 'assets/img/Avatar.svg';
import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import ProfileImage from './ProfileImage';

describe('ProfileImage', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const messageItem = renderWithProviders(<ProfileImage src={Avatar} size={1.5} />, {});
    expect(messageItem).toMatchSnapshot();
  });
});
