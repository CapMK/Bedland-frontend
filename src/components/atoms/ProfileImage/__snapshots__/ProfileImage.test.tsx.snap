// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileImage > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  border-radius: 50%;
  height: 1.5rem;
  width: 1.5rem;
}

<body>
    <div>
      <img
        alt="profile"
        class="c0"
        size="1.5"
        src="/src/assets/img/Avatar.svg"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  border-radius: 50%;
  height: 1.5rem;
  width: 1.5rem;
}

<div>
    <img
      alt="profile"
      class="c0"
      size="1.5"
      src="/src/assets/img/Avatar.svg"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
